addons/sys_core/fnc_switchChannelFast.sqf

```diff
@@ -35,7 +35,20 @@ if (_isManpack == 0 || {_isRackRadio}) then {
             _channel = ((_channel + _dir) max (_currentBlock*16 + 1)) min ((_currentBlock + 1)*16);
         };
         case "ACRE_PRC152": {
-            _channel = (_channel + _dir) min 5;
+            if (_dir == 1) then {
+                if (_channel < 99) then {
+                    _channel = _channel + 1;
+                } else {
+                    _channel = 0;
+                };
+            } else {
+                if (_channel > 1) then {
+                    _channel = _channel - 1;
+                } else {
+                    // Go to the last preset
+                    _channel = 99;
+                };
+            };
```

Comment:
```suggestion
            _channel = (_channel + _dir + 100) % 100;
```

Reply:
Unfortunately it doesn't work when stepping backward over Ch1->Ch99. It just remains on Ch1, because:
`(1 + (-1) + 100) % 100 = 100 % 100 = 0 (expected 99)`

Reply:
Not working on Ch99->Ch1, because:
`(99 + (+1) + 100) % 100 = 200 % 100 = 0`, making the `if` then kick in.

Reply:
What are you trying? The suggestion was updated, it is now `(_channel + _dir + 98) % 99 + 1` which works for me at [99,+1] and [1,-1]

Reply:
I must have hallucinated seeing the following code in the suggestion.
```sqf
_channel = (_channel + _dir + 100) % 100;
if (_channel == 0) then {
    _channel = 99;
};
```

Reply:
`(_channel + _dir + 98) % 99 + 1` works 👍🏻

Reply:
It was there for a short bit, but was replaced not too long after

Reply:
Awesome, I kept debugging the first draft until I wrote my comment 😆